### PR TITLE
SCL-9579

### PR DESCRIPTION
**Pull request by neowulf**
Initial PR comments can be found here: https://github.com/JetBrains/intellij-scala/pull/348

### CODE REVIEW

File: test/org/jetbrains/plugins/scala/codeInsight/intentions/booleans/ReplaceEqualsOrEqualityInMethodCallExprIntentionTest.scala [hunk 1]
```diff
@@ -12,16 +12,51 @@ import org.jetbrains.plugins.scala.codeInsight.intentions.ScalaIntentionTestBase
...
+
+  def testReplaceEqualityInExpression() = {
+    val text = "(x.eq<caret>uals(y)).toString"
+    val resultText = "(x <caret>== y).toString"
+
+    doTest(text, resultText)
+  }
+
+  def testReplaceEqualityInExpression_1() = {
+    val text = "x.eq<caret>uals(y).toString"
```
Comment (niktrop) [inline test/org/jetbrains/plugins/scala/codeInsight/intentions/booleans/ReplaceEqualsOrEqualityInMethodCallExprIntentionTest.scala:58]: Parentheses should be added in this case.

File: src/org/jetbrains/plugins/scala/codeInsight/intention/booleans/ReplaceEqualsOrEqualityInMethodCallExprIntention.scala [hunk 2]
```diff
@@ -29,43 +35,70 @@ class ReplaceEqualsOrEqualityInMethodCallExprIntention extends PsiElementBaseInt
...
+    val size = {
+      if (desiredOper == "==") {
+        newMethodCallExpr.asInstanceOf[ScInfixExprImpl].operation.nameId.getTextRange.getStartOffset - newMethodCallExpr.getTextRange.getStartOffset
+      } else {
+        newMethodCallExpr.asInstanceOf[ScMethodCall].getInvokedExpr.asInstanceOf[ScReferenceExpression].nameId.
+          getTextRange.getStartOffset - newMethodCallExpr.getTextRange.getStartOffset
+      }
+    }
 
     inWriteAction {
```
Comment (niktrop) [inline src/org/jetbrains/plugins/scala/codeInsight/intention/booleans/ReplaceEqualsOrEqualityInMethodCallExprIntention.scala:76]: You need to use `replaceExpression` here, as we discussed before.

**Comment by niktrop** [comment]
Initial ticket was about adding an inspection in addition to the intention. Are you going to implement it? I may merge this PR as it is when you fix `testReplaceEqualityInExpression_1`.

**Comment by neowulf33** [comment]
Yikes, true. I will try refactoring using `InspectionBasedIntention`.

**Comment by niktrop** [comment]
I've merged your commits with some minor changes: https://github.com/JetBrains/intellij-scala/commit/834ca1b55a362ccb0fe10721dc3149cb06dc6b81
I'm closing this PR now, please create another one if you are going to implement a corresponding  inspection.

**Pull request closed without merging**